main/src/ca/uwaterloo/flix/language/phase/Parser2.scala

```diff
@@ -3412,8 +3412,8 @@ object Parser2 {
   def syntaxTreeToDebugString(tree: SyntaxTree.Tree, nesting: Int = 1): String = {
     s"${tree.kind}${
       tree.children.map {
-        case SyntaxTree.Child.TokenChild(token) => s"\n${"  " * nesting}'${token.text}'"
-        case SyntaxTree.Child.TreeChild(tree) => s"\n${"  " * nesting}${syntaxTreeToDebugString(tree, nesting + 1)}"
+        case token: Token => s"\n${"  " * nesting}'${token.text}'"
```

Comment:
We try to avoid type based matches. You can still just match on the constructor, right?